pkg/controllers/clusterresourceplacement/placement_status_test.go

```diff
@@ -3123,405 +3163,1800 @@ func TestSetPlacementStatus(t *testing.T) {
+								Reason:             condition.ScheduleSucceededReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceWorkSynchronizedConditionType),
+								Reason:             condition.WorkSynchronizedReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourcesDiffReportedConditionType),
+								Reason:             condition.DiffReportedStatusTrueReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+						},
+					},
+					{
+						ClusterName:                        "member-2",
+						ApplicableClusterResourceOverrides: []string{"o-1", "o-2"},
+						ApplicableResourceOverrides: []fleetv1beta1.NamespacedName{
+							{
+								Name:      "override-1",
+								Namespace: "override-ns",
+							},
+							{
+								Name: "override-2",
+							},
+						},
+						Conditions: []metav1.Condition{
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceOverriddenConditionType),
+								Reason:             condition.OverriddenSucceededReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceRolloutStartedConditionType),
+								Reason:             condition.RolloutStartedReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceScheduledConditionType),
+								Reason:             condition.ScheduleSucceededReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceWorkSynchronizedConditionType),
+								Reason:             condition.WorkSynchronizedReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionFalse,
+								Type:               string(fleetv1beta1.ResourceBindingDiffReported),
+								Reason:             condition.DiffReportedStatusFalseReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			crp := tc.crp
```

Comment:
hmm it's confusing, why not just using tc.crp directly?

duplicate crp definition in line 63,

Reply:
Adjusted.